Client/Writes/PointData.cs

```diff
@@ -419,12 +498,112 @@ private void EscapeValue(StringBuilder sb, string value)
             }
         }
 
+        /// <summary>
+        /// Determines whether [is not defined] [the specified value].
+        /// </summary>
+        /// <param name="value">The value.</param>
+        /// <returns>
+        ///   <c>true</c> if [is not defined] [the specified value]; otherwise, <c>false</c>.
+        /// </returns>
         private bool IsNotDefined(object value)
         {
             return value == null
                    || (value is double d && (double.IsInfinity(d) || double.IsNaN(d)))
                    || (value is float f && (float.IsInfinity(f) || float.IsNaN(f)));
         }
+
+        /// <summary>
+        /// Determines whether the specified <see cref="System.Object" />, is equal to this instance.
+        /// </summary>
+        /// <param name="obj">The <see cref="System.Object" /> to compare with this instance.</param>
+        /// <returns>
+        ///   <c>true</c> if the specified <see cref="System.Object" /> is equal to this instance; otherwise, <c>false</c>.
+        /// </returns>
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as PointData);
+        }
+
+        /// <summary>
+        /// Indicates whether the current object is equal to another object of the same type.
+        /// </summary>
+        /// <param name="other">An object to compare with this object.</param>
+        /// <returns>
+        /// true if the current object is equal to the <paramref name="other">other</paramref> parameter; otherwise, false.
+        /// </returns>
+        public bool Equals(PointData other)
+        {
+            if (other == null)
+                return false;
+            var otherTags = other._tags;
+            bool result = _tags.Count == otherTags.Count &&
+                           _tags.All(pair => otherTags.ContainsKey(pair.Key) &&
+                                             otherTags[pair.Key] == pair.Value);
+            var otherFields = other._fields;
+            result = result && _fields.Count == otherFields.Count &&
+                           _fields.All(pair => otherFields.ContainsKey(pair.Key) &&
+                                             object.Equals(otherFields[pair.Key], pair.Value));
```

Comment:
Please remove redundant qualifier => `Equals(otherFields[pair.Key], pair.Value));`

Reply:
The `object` qualifier is unnecessary.